NumLib/Fem/ShapeFunction/ShapeQuad4-impl.h

```diff
@@ -16,23 +16,24 @@ namespace NumLib
 template <class T_X, class T_N>
 void ShapeQuad4::computeShapeFunction(const T_X &r, T_N &N)
 {
-    N[0] = (1.0 + r[0]) * (1.0 + r[1]) / 4;
-    N[1] = (1.0 - r[0]) * (1.0 + r[1]) / 4;
-    N[2] = (1.0 - r[0]) * (1.0 - r[1]) / 4;
-    N[3] = (1.0 + r[0]) * (1.0 - r[1]) / 4;
+    N[0] = (1.0 - r[0]) * (1.0 - r[1]) / 4;
+    N[1] = (1.0 + r[0]) * (1.0 - r[1]) / 4;
+    N[2] = (1.0 + r[0]) * (1.0 + r[1]) / 4;
+    N[3] = (1.0 - r[0]) * (1.0 + r[1]) / 4;
 }
```

Comment:
Yes,   N_u and N_p shape functions must correspond. It means that for a geometrical element type, the node ordering rule for all finite element types must be the same.
In summary,  the problem is caused by
1. quad4 and quad8 has  the different node ordering.
2. quad4 and quad9 has the same node order but quad9 got wrong modification in grad N after copying. Got correct results by removing the modification in grad N of quad9 without change the order of N.

I am going to prepare a test to check grad N and whether the same order rule are applied for both linear and high order   elements within a geometrical element type.